Fixes to the new inspector

guilhermefelipecgs opened the pull request:
Fix #19008, Fix  #19072.
- Fix inspector dock not updating tree for main resource;
- Also fix "docks/property_editor/auto_refresh_interval" used by MultiNodeEdit, it wasn't working.

## Update:
Fix https://github.com/godotengine/godot/issues/9890#issuecomment-394844248.
Before:
![out mp4](https://user-images.githubusercontent.com/1387165/41000087-3ec76676-68e3-11e8-8884-ccd20be770f9.gif)
After:
![out mp4](https://user-images.githubusercontent.com/1387165/41182469-3357f43e-6b4c-11e8-9b6a-006713435ca7.gif)
This happens when you try to slide to a value that the property doesn't allow (because of the minimum size of the children is "dynamic"), in this case the `margin_left` can't go above 15, but here it's set that can range from -4096 to 4096.
https://github.com/godotengine/godot/blob/66871322b99436990daec5a70c09620619430b8d/scene/gui/control.cpp#L2868
and the fix was just pass "anchor" on `_change_notify`.

guilhermefelipecgs commented on editor/editor_inspector.cpp:
```diff
@@ -1709,6 +1709,8 @@ void EditorInspector::_edit_set(const String &p_name, const Variant &p_value, bo
 		}
 	}
 
+	changing++;
```
I moved this variable here to fix multiedit node and all other flows we can have inside this logic too.

guilhermefelipecgs commented:
I removed:
- signal: `multiple_properties_changed`
- method: `add_property_editor_for_multiple_properties`, `_multiple_properties_changed`

because no one was using.

Maybe @reduz wants to review.

vnen commented:
Is there anything missing for this? #19008 is an extremely annoying bug.

guilhermefelipecgs commented:
> 19008 is an extremely annoying bug.

Yeah, this is annoying me too.

This pr is complete and waiting to merge, I just pointed out that I am removing an unused method that @reduz might be planning to use in the future, but I believe that since he didn't change the inspector again, he probably didn't intend to use it at all.

vnen commented:
I'm not sure if he actually forgot an `emit_signal()` somewhere.

reduz commented:
Please dont remove multiple properties changed, it will be used in the reworked anchor/margin editor

guilhermefelipecgs committed:
Fixes to the new inspector

- Fix inspector dock not updating tree for main resource;
- Fixes the inspector input text reverted during typing;
- Add method bind for "refresh" used by MultiNodeEdit;

guilhermefelipecgs commented:
Ok, changes made.
